closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -0,0 +1,1053 @@
+            if (debug != null) {
+                debug.println("Loading restricted security properties.");
+            }
+
+            descName = parseProperty(securityProps.getProperty(propsPrefix + ".desc.name"));
+            descNumber = parseProperty(securityProps.getProperty(propsPrefix + ".desc.number"));
+            descPolicy = parseProperty(securityProps.getProperty(propsPrefix + ".desc.policy"));
+            descSunsetDate = parseProperty(securityProps.getProperty(propsPrefix + ".desc.sunsetDate"));
+
+            jdkTlsDisabledNamedCurves = parseProperty(
+                    securityProps.getProperty(propsPrefix + ".tls.disabledNamedCurves"));
+            jdkTlsDisabledAlgorithms = parseProperty(
+                    securityProps.getProperty(propsPrefix + ".tls.disabledAlgorithms"));
+            jdkTlsDphemeralDHKeySize = parseProperty(
+                    securityProps.getProperty(propsPrefix + ".tls.ephemeralDHKeySize"));
+            jdkTlsLegacyAlgorithms = parseProperty(
+                    securityProps.getProperty(propsPrefix + ".tls.legacyAlgorithms"));
+            jdkCertpathDisabledAlgorithms = parseProperty(
+                    securityProps.getProperty(propsPrefix + ".jce.certpath.disabledAlgorithms"));
+            jdkSecurityLegacyAlgorithm = parseProperty(
+                    securityProps.getProperty(propsPrefix + ".jce.legacyAlgorithms"));
+            keyStoreType = parseProperty(
+                    securityProps.getProperty(propsPrefix + ".keystore.type"));
+            keyStore = parseProperty(
+                    securityProps.getProperty(propsPrefix + ".javax.net.ssl.keyStore"));
+
+            jdkSecureRandomProvider = parseProperty(
+                    securityProps.getProperty(propsPrefix + ".securerandom.provider"));
+            jdkSecureRandomAlgorithm = parseProperty(
+                    securityProps.getProperty(propsPrefix + ".securerandom.algorithm"));
+
+            if (debug != null) {
+                debug.println("Loaded restricted security properties.");
+            }
+        }
+
+        /**
+         * Load security constraints with type, algorithm, attributes.
+         *
+         * Example:
+         * RestrictedSecurity1.jce.provider.1 = SUN [{CertPathBuilder, PKIX, *},
+         * {Policy, JavaPolicy, *}, {CertPathValidator, *, *}].
+         */
+        private void initConstraints() {
+            for (int pNum = 1; pNum <= providersSimpleName.size(); pNum++) {
+                String providerName = providersSimpleName.get(pNum - 1);
+                String providerInfo = securityProps
+                        .getProperty(propsPrefix + ".jce.provider." + pNum);
+
+                if (debug != null) {
+                    debug.println("Loading constraints for security provider: " + providerName);
+                }
+
+                // Check if the provider has constraints
+                if (providerInfo.indexOf('[') < 0 && providerInfo.indexOf(']') < 0) {
```

Comment:
Please finish the sentence with a period.
As we've already ensure the brackets are balanced, it's sufficient to check for just `[`:
```java
                if (providerInfo.indexOf('[') < 0) {
```